app/components/chat_component.rb

```diff
@@ -0,0 +1,45 @@
+class ChatComponent < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :user, :conversation, :messages, :send_message_id
+
+  def initialize(user:, conversation:, messages:)
+    @user = user
+    @conversation = conversation
+    @messages = messages
+    @send_message_id = SecureRandom.uuid
+    self.class.stream_for(conversation, :handle_received)
```

Comment:
looks hacky but couldn't have it outside the `initialize` function since it depends on `conversation`